src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -385,8 +385,15 @@ private IntPtr ObjectToHandle(Object obj)
             return handle;
         }
 
+        // We don't have System.TypedReference but RyuJIT can ask for a classhandle for it anyway.
+        // This is the classhandle we give out. RyuJIT will only use it for comparisons and those
+        // will always be false as expected. It's not valid to pass it back to us.
+        private const int FakeTypedReferenceTypeHandle = 0xBEF;
+
         private Object HandleToObject(IntPtr handle)
         {
+            Debug.Assert(handle != (IntPtr)FakeTypedReferenceTypeHandle);
```

Comment:
Would you mind if I just file an issue on that and go with this? (Plus the change to make it throw.)

I'm trying to wrap up the shared generics work today and I have a RyuJIT package that has everything needed. Having to wait for another one will make it impossible to wrap it up today. Plus the next RyuJIT package will also have a breaking JitInterface change I would need to deal with on top of this. @sergiy-k wants me to start reflection work ASAP.